tests/ulyanov_et_al_2016/test_modules.py

```diff
@@ -120,3 +167,187 @@ def test_get_activation_module(subtests):
+    kernel_size = 3
+    stride = 1
+    padding = (1, 1, 1, 1)
+    conv_block = modules.UlyanovEtAl2016ConvBlock(
+        in_channels, out_channels, kernel_size, stride=stride
+    )
+
+    assert isinstance(type(conv_block), type(modules.SequentialWithOutChannels))
+
+    with subtests.test("modules"):
+        assert len(conv_block) == 4
+        assert isinstance(type(conv_block[0]), type(nn.ReflectionPad2d))
+        with subtests.test("conv_module"):
+            assert isinstance(type(conv_block[1]), type(nn.Conv2d))
+            assert conv_block[1].stride == to_2d_arg(stride)
+            assert conv_block[1].padding == to_2d_arg(0)
+        assert isinstance(type(conv_block[2]), type(nn.InstanceNorm2d))
+        assert isinstance(type(conv_block[3]), type(nn.ReLU))
+
+    with subtests.test("padding"):
+        assert conv_block[0].padding == padding
+
+
+def test_UlyanovEtAl2016ConvSequence(subtests):
+    in_channels = 3
+    out_channels = 6
+    kernel_size = 3
+    conv_sequence = modules.UlyanovEtAl2016ConvSequence(in_channels, out_channels)
+
+    assert isinstance(type(conv_sequence), type(modules.SequentialWithOutChannels))
+
+    with subtests.test("modules"):
+        assert len(conv_sequence) == 3
+        for i in range(len(conv_sequence)):
+            assert isinstance(
+                type(conv_sequence[i]), type(modules.UlyanovEtAl2016ConvBlock)
+            )
+            assert (
+                conv_sequence[i][1].in_channels == in_channels
+                if i == 0
+                else out_channels
+            )
+            assert conv_sequence[i].out_channels == out_channels
+            assert (
+                conv_sequence[i][1].kernel_size == to_2d_arg(kernel_size)
+                if i < len(conv_sequence) - 1
+                else to_2d_arg(1)
+            )
+
+
+def test_UlyanovEtAl2016JoinBlock(subtests, input_image):
+    in_channels = extract_num_channels(input_image)
+    channel_dim = 1
+    configs = (
+        ((in_channels, in_channels), ("block1", "block_2")),
+        ((in_channels, in_channels), None),
+    )
+    for branch_in_channels, names in configs:
+        block = modules.UlyanovEtAl2016JoinBlock(
+            branch_in_channels, names=names, channel_dim=channel_dim
+        )
+
+        with subtests.test("names"):
+            assert (
+                tuple(block._modules.keys()) == names
+                if names is not None
+                else ("0", "1")
+            )
+        with subtests.test("modules"):
+            assert len(block.norm_modules) == len(branch_in_channels)
+
+        with subtests.test("out_channels"):
+            assert block.out_channels == sum(branch_in_channels)
+
+        with subtests.test("channel_dim"):
+            assert block.channel_dim == channel_dim
+
+        with subtests.test("forward"):
+            inputs = (input_image,) * 2
+            output_image = block(*inputs)
+            assert isinstance(output_image, torch.Tensor)
```

Comment:
You need to actually check the output for example with [`instance_norm`](https://pytorch.org/docs/stable/nn.functional.html?highlight=instance_norm#torch.nn.functional.instance_norm).